doc/source/catalog_generation.rst

```diff
@@ -23,67 +31,153 @@ It should also be stressed here that the point and segment photometry source lis
-Pixels in the immediate vicinity of a given masked region may also be impacted to some degree. As we cannot
-be fully certain that these pixels are or are not impacted, or to the degree of the impact, they are all
-flagged.
-
-1.2.2: Detection Image Background Subtraction
-""""""""""""""""""""""""""""""""""""""""""""""""
-To ensure optimal source detection, the multi-filter detection image is background-subtracted. We computed a
-2-dimensional background image using the `photutils.background.Background2d <https://photutils.readthedocs.io/en/stable/api/photutils.background.Background2D.html>`_ Astropy tool. This algorithm uses
-sigma-clipped statistics to determine background and RMS values across the image. An initial low-resolution
+1.2: Generation of Pixel Masks
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Every multi-filter, detector-level drizzle-combined image is associated with a boolean footprint mask which 
+defines the illuminated (True) and non-illuminated (False) portions of the image based upon its constituent 
+exposures and the corresponding WCS solution.  The boundary of the illuminated portion
+is iteratively eroded or contracted to minimize the impact of regions where signal
+quality is known to be degraded, and thereby, could affect source identification and subsequent 
+photometric measurements.  The erosion depth is approximately ten pixels and is done by using the 
+`ndimage.binary_erosion <https://docs.scipy.org/doc/scipy/reference/generated/scipy.ndimage.binary_erosion.html>`_ scipy tool.
+For computational convenience, an inverse footprint mask is also available for functions
+which utilize masks to indicate pixels which should be ignored during processing of the 
+input data.
+
+1.3: Detection Image Background Determination
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+For consistency, the same background and background RMS images are used by both the point and
+segment algorithms.
+To ensure optimal source detection, the multi-filter detection image must be background-subtracted. 
+In order to accommodate the different types of detectors, disparate signal levels, and highly varying 
+astronomical image content, three background computations are used as applicable.  The first category 
+of background definition is a special case situation, and if it is found to be applicable to the detection 
+image, the background and background RMS images are defined and no further background evaluation is done. 
+
+It has been observed that some background regions of ACS SBC drizzle-combined 
+detection images, *though the evaluation is done for all instrument detection images*,
+are measured to have values identically
+equal to zero.  If the number of identically zero pixels in the footprint portion of the detection image
+exceeds a configurable percentage threshold value (default is 25%), then a two-dimensional background image 
+is constructed, **Background Algorithm 1**, and set to the value of zero. Its companion 
```

Comment:
The terms 'Background Algorithm 1' and so on unfortunately do not help the user understand what the code is doing.  It would actually be more explicit to give them new labels; such as `Zero Background Algorithm`, `Constant Background Algorithm` and `Conformal Background Algorithm`.

Reply:
Good idea. Done as bd066eb.